lib/rspec-puppet/support.rb

```diff
@@ -224,7 +225,18 @@ def stub_facts!(facts)
     end
 
     def build_catalog(*args)
-      @@cache[args] ||= self.build_catalog_without_cache(*args)
+      unless @@cache.has_key? args
+        # Keep only the most recently added 16 entries to prevent high memory consumption
+        expire_cache(@@cache, @@cache_lra, 15)
```

Comment:
I've moved the line down so that it's below the new addition to the cache, meaning `16` can be passed into `expire_cache`.  Since it's the most recently added, there's no danger of it being immediately expired again.